agent/xds/xdscommon/xdscommon_oss_test.go

```diff
@@ -37,60 +37,70 @@ func TestMakePluginConfiguration_TerminatingGateway(t *testing.T) {
+	expected := map[api.CompoundServiceName][]ExtensionConfiguration{
+		apiService:   {},
+		cacheService: {},
+		dbService:    {},
+		webService: {
+			{
+				EnvoyExtension: api.EnvoyExtension{
+					Name: structs.BuiltinAWSLambdaExtension,
+					Arguments: map[string]interface{}{
+						"ARN":                "lambda-arn",
+						"PayloadPassthrough": true,
+						"Region":             "us-east-1",
+					},
+				},
+				ServiceName: webService,
+				Upstreams: map[api.CompoundServiceName]UpstreamData{
+					apiService: {
+						SNI: map[string]struct{}{
+							"api.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul": {},
+						},
+						EnvoyID:           "api",
+						OutgoingProxyKind: "terminating-gateway",
+					},
+					cacheService: {
+						SNI: map[string]struct{}{
+							"cache.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul": {},
+						},
+						EnvoyID:           "cache",
+						OutgoingProxyKind: "terminating-gateway",
+					},
+					dbService: {
+						SNI: map[string]struct{}{
+							"db.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul": {},
 						},
+						EnvoyID:           "db",
+						OutgoingProxyKind: "terminating-gateway",
+					},
+					webService: {
+						SNI: map[string]struct{}{
+							"canary1.web.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul": {},
+							"canary2.web.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul": {},
+							"web.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul":         {},
+						},
+						EnvoyID:           "web",
+						OutgoingProxyKind: "terminating-gateway",
 					},
 				},
-			},
-			apiService: {
 				Kind: api.ServiceKindTerminatingGateway,
 			},
-			cacheService: {
-				Kind: api.ServiceKindTerminatingGateway,
-			},
-			dbService: {
-				Kind: api.ServiceKindTerminatingGateway,
-			},
-		},
-		SNIToServiceName: map[string]api.CompoundServiceName{
-			"api.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul":         apiService,
-			"cache.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul":       cacheService,
-			"db.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul":          dbService,
-			"web.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul":         webService,
-			"canary1.web.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul": webService,
-			"canary2.web.default.dc1.internal.11111111-2222-3333-4444-555555555555.consul": webService,
-		},
-		EnvoyIDToServiceName: map[string]api.CompoundServiceName{
-			"web":   webService,
-			"db":    dbService,
-			"cache": cacheService,
-			"api":   apiService,
 		},
 	}
 
-	require.Equal(t, expected, MakePluginConfiguration(snap))
+	require.Equal(t, expected, GetExtensionConfigurations(snap))
 }
 
-func TestMakePluginConfiguration_ConnectProxy(t *testing.T) {
+func TestGetExtensionConfigurations_ConnectProxy(t *testing.T) {
```

Comment:
It might be helpful to add a case here where an upstream is registered to a terminating gateway. We test this through the golden tests, but it feels like testing it at the source could be useful too.

Reply:
Good call, added!